UPGRADE-4.3.md

```diff
@@ -55,6 +55,35 @@ EventDispatcher
 ---------------
 
  * The signature of the `EventDispatcherInterface::dispatch()` method should be updated to `dispatch($event, string $eventName = null)`, not doing so is deprecated
+
+   You can use `Symfony\Component\EventDispatcher\LegacyEventDispatcherProxy` and swap arguments of `dispatch()` to ease upgrade:
```

Comment:
I think we should say *when* this is useful, so that ppl don't always use the proxy but more often update the way they use the dispatcher directly.

Reply:
Yes, using this proxy still generate warnings:

> [php] User Deprecated: Calling the "Symfony\Component\EventDispatcher\EventDispatcherInterface::dispatch()" method with the event name as first argument is deprecated since Symfony 4.3, pass it second and provide the event object first instead. ["exception" => ErrorException { …}] []

I'm trying to find how to switch to the new `EventDispatcher` without using the proxy.

Reply:
I rewrote this part by using the explanation in #28920